File: 1_1.py
with open("1_1_input.txt", "r") as file:
    lines = file.readlines()
    highTotal = [0,0,0]
    subTotal = 0
    for line in lines:
        if line != '\n':
            subTotal += int(line)
        else:
            if subTotal > highTotal[0]:
                highTotal[2] = highTotal[1]
                highTotal[1] = highTotal[0]
                highTotal[0] = subTotal
            elif subTotal > highTotal[1]:
                highTotal[2] = highTotal[1]
                highTotal[1] = subTotal
            elif subTotal > highTotal[2]:
                highTotal[2] = subTotal            
            subTotal = 0
    print(sum(highTotal))

File: 1_1.rs
fn main() {
    let string = std::fs::read_to_string("1_1_input.txt").expect("Open failed");

    let iter = string.split("\n\n");
    let mut vec: Vec<i32> = Vec::new();
    iter.into_iter().for_each(|s| {
        let elf: Vec<i32> = s.trim().split("\n").map(|x| {x.parse::<i32>().unwrap()}).collect();
        vec.push(elf.iter().sum::<i32>());
    });
    vec.sort();
    vec.reverse();
    let mut sum = 0;
    for i in 0..3 {
        sum += vec[i];
    }
    println!("{}", sum);
}